### PR TITLE
Add hand-off target upsert

### PR DESCRIPTION
**Pull request by andrewgarner**
Part of GRA-430

### CODE REVIEW

**Comment by andrewgarner** [comment]
* **#19** <a href="https://app.graphite.dev/github/pr/gradientlabs-ai/go-client/19?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @andrewgarner and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: hand_off_target_upsert.go
Comment (twindebank) [inline hand_off_target_upsert.go:9]: Is this validated at all? I wonder if we should validate it (in the backend), and perhaps also provide an example

**Review by twindebank — COMMENTED** [review]
LGTM but I'm going to hold off approving as I don't know if there are any nuances of this repo to be aware of, will defer to @nlathia / @boxofrad

File: hand_off_target_upsert.go
Comment (andrewgarner) [reply hand_off_target_upsert.go:9]: It's validated on the server — https://github.com/gradientlabs-ai/wearegradient/blob/8530c9f0f83594f0214a7120736618456f202dd2/apis/public-api/hand_off_target_upsert.go#L29. I'll add some additional context about what is valid [like we do in the api-docs](https://github.com/gradientlabs-ai/api-docs/blob/cae2c7e7ced47aa49f9fd1a27e1d70c267af51c7/source/index.html.md?plain=1#L175).

**Review by nlathia — APPROVED** [review]

**Pull request merged**
